src/models/github-server-app.test.ts

```diff
@@ -128,6 +135,92 @@ describe("GitHubServerApp", () => {
 					}
 				});
 			});
+
+			describe("GitHubServerApp update", () => {
+				const gitHubBaseUrl = "http://myinternalinstance.com";
+				const installationId = 42;
+				const uuid = "c97806fc-c433-4ad5-b569-bf5191590be2";
+				const clientIdWithTypo = "lvl.1n23j12389wndd";
+				const originalWebhookSecret = "anothersecret";
+				const originalPrivateKey = "privatekey";
+				let gitHubServerApp;
+
+				beforeEach(async () => {
+					gitHubServerApp = await GitHubServerApp.create({
+						uuid,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: clientIdWithTypo,
+						gitHubClientSecret: "secret",
+						webhookSecret: "anothersecret",
+						privateKey: "privatekey",
+						installationId
+					});
+				});
+
+				it("should update GitHub app when uuid is found", async () => {
+					const originalApp = await GitHubServerApp.findForUuid(uuid);
+					expect(originalApp && originalApp.gitHubClientId).toEqual(clientIdWithTypo);
+					expect(originalApp && originalApp.webhookSecret).toEqual(`encrypted:${originalWebhookSecret}`);
+					expect(originalApp && originalApp.privateKey).toEqual(`encrypted:${originalPrivateKey}`);
+
+					await GitHubServerApp.updateGitHubAppByUUID({
+						uuid,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: "lvl.1n23j12389wnde",
+						gitHubClientSecret: "secret",
+						webhookSecret: "anewsecret",
+						privateKey: "privatekeyversion2",
+						installationId
+					});
+
+					const updatedApp = await GitHubServerApp.findForUuid(uuid);
+					expect(updatedApp && updatedApp.uuid).toEqual(gitHubServerApp.uuid);
+					expect(updatedApp && updatedApp.gitHubClientId).toEqual("lvl.1n23j12389wnde");
+					expect(updatedApp && updatedApp.webhookSecret).toEqual("anewsecret");
+					expect(updatedApp && updatedApp.privateKey).toEqual("privatekeyversion2");
+				});
+
+				it("should not update GitHub app when uuid is not found", async () => {
+					const mismatchedUUID = "c97806fc-c433-4ad5-b569-bf5191590ba9";
+
+					await GitHubServerApp.updateGitHubAppByUUID({
+						uuid: mismatchedUUID,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: "lvl.1n23j12389wnde",
+						gitHubClientSecret: "secret",
+						webhookSecret: "anewsecret",
+						privateKey: "privatekeyversion2",
+						installationId
+					});
+
+					const updatedApp = await GitHubServerApp.findForUuid(mismatchedUUID);
+					expect(updatedApp && updatedApp.uuid).not.toEqual(gitHubServerApp.uuid);
+					expect(updatedApp && updatedApp.gitHubClientId).not.toEqual("lvl.1n23j12389wnde");
+					expect(updatedApp && updatedApp.webhookSecret).not.toEqual("anewsecret");
```

Comment:
Hmm, that's weird, shouldn't it be `updatedApp.decrypt("webhookSecret")` inorder to get the origin value?

Reply:
Alright, seems like a bug in the base GitHubServerApp model. Let me have a look.

Reply:
`updatedApp.webhookSecret` is always `encrypted:xxx`, so never equal to anything `anewsecret` or `anothersecret`. The test itself is invalid.
I suggest change it to:
`expect(await updatedApp.decyrpt("webhookSecret")).toBe(ORIGIN_WEBHOOK_SECRET);`

Reply:
What do we set ORIGIN_WEBHOOK_SECRET to?

Reply:
`ORIGIN_WEBHOOK_SECRET` is a const var that we use above to create the existing app in `beforeEach`. 
But this is /nit, just for better readability on the test.

Only thing need to change is, to be a valid test, I think we should use `updatedApp.decrypt("webhookSecret")).toEqual` instead of `updatedApp.webhookSecret).not.toEqual`